sdk/storage/azure-storage-file-share/azure/storage/fileshare/_models.py

```diff
@@ -787,29 +999,28 @@ class CopyProperties(DictMixin):
             aborted:
                 Copy was ended by Abort Copy File.
             failed:
-                Copy failed. See copy_status_description for failure details.
-    :ivar str progress:
-        Contains the number of bytes copied and the total bytes in the source in the last
+                Copy failed. See copy_status_description for failure details."""
+    progress: Optional[str] = None
+    """Contains the number of bytes copied and the total bytes in the source in the last
         attempted Copy File operation where this file was the destination file. Can show
-        between 0 and Content-Length bytes copied.
-    :ivar datetime completion_time:
-        Conclusion time of the last attempted Copy File operation where this file was the
+        between 0 and Content-Length bytes copied."""
+    datetime: Optional["datetime"] = None
+    """Conclusion time of the last attempted Copy File operation where this file was the
         destination file. This value can specify the time of a completed, aborted, or
-        failed copy attempt.
-    :ivar str status_description:
-        Only appears when x-ms-copy-status is failed or pending. Describes cause of fatal
-        or non-fatal copy operation failure.
-    :ivar bool incremental_copy:
-        Copies the snapshot of the source file to a destination file.
+        failed copy attempt."""
+    status_description: Optional[str] = None
+    """Only appears when x-ms-copy-status is failed or pending. Describes cause of fatal
+        or non-fatal copy operation failure."""
+    incremental_copy: Optional[bool] = None
+    """Copies the snapshot of the source file to a destination file.
         The snapshot is copied such that only the differential changes between
-        the previously copied snapshot are transferred to the destination
-    :ivar datetime destination_snapshot:
-        Included if the file is incremental copy or incremental copy snapshot,
+        the previously copied snapshot are transferred to the destination."""
+    destination_snapshot: Optional["datetime"] = None
```

Comment:
This line is causing a really interesting error that is fixed by removing `destination_snapshot` and its associated docstring:

```python3
azure/storage/fileshare/_models.py:1018: error: Variable "azure.storage.fileshare._models.CopyProperties.datetime" is not valid as a type  [valid-type]
azure/storage/fileshare/_models.py:1018: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
```

Any suggestions here?

Reply:
Here are two ideas:
1. Simply put a `#type: ignore [valid-type]` to the end
2. Move `datetime` class variable after `destination_snapshot` to avoid type collision

Reply:
Looks like option 2 did make the errors disappear. Let me know you prefer the first or some other solution!

Reply:
Will defer to Jacob here